sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -751,6 +751,25 @@ class HiveDDLSuite
     }
   }
 
+  test("SPARK-23057: SET LOCATION should change the path of partition in table") {
+    withTable("boxes") {
+      sql("CREATE TABLE boxes (height INT, length INT) PARTITIONED BY (width INT)")
+      sql("INSERT OVERWRITE TABLE boxes PARTITION (width=4) SELECT 4, 4")
+      val expected = "/path/to/part/ways"
+      sql(s"ALTER TABLE boxes PARTITION (width=4) SET LOCATION '$expected'")
+      val catalog = spark.sessionState.catalog
```

Comment:
This is close to the method body checkPath(). Is it possible to find a common place for the method and call the method from here too? Like creating an object close the DDLSuite. What is your opinion?

Reply:
Good idea, I will try